assets/js/src/core/modules/element/dynamic-types/defintinitions/objects/data-related/components/object-brick/object-brick.tsx

```diff
@@ -33,12 +34,12 @@ export const ObjectBrick = (props: ObjectBrickProps): React.JSX.Element => {
   const currentValue = Form.useWatch(props.name, form)
 
   const dropdownItems: DropdownMenuProps['items'] = props.allowedTypes.map((type) => {
-    const hasType = currentValue?.some((fieldValue) => fieldValue.type === type) ?? false
+    const hasType = (isArray(currentValue) && currentValue.some((fieldValue) => fieldValue.type === type)) ?? false
```

Comment:
Ok for now but @vin0401 needs to check the object bricks logic in general - data structure from API and frontend seem to not match anymore